sqlglot/parser.py

```diff
@@ -2049,7 +2049,23 @@ def _parse_function(self, functions=None):
             args = self._parse_csv(self._parse_lambda)
 
             if function:
-                this = function(args)
+                params = None
+
+                # Clickhouse supports function calls like foo(x, y)(z), so for these we need to also parse the
+                # second parameter list (i.e. "(z)") and the corresponding function will receive both arg lists.
+                if function.__code__.co_name == "<lambda>" and function.__code__.co_argcount == 2:
+                    index = self._index
+                    self._match_r_paren()
```

Comment:
i don't like doing the retreat here, can we do this another way?

Reply:
fixed.